src/coupling/hc_explicit_sequential.cc

```diff
@@ -117,14 +113,15 @@ HC_ExplicitSequential::HC_ExplicitSequential(Input::Record in_record)
     // TODO: optionally setup transport objects
     Iterator<AbstractRecord> it = in_record.find<AbstractRecord>("secondary_equation");
     if (it) {
-    	transport_reaction = (*it).factory< AdvectionProcessBase, Mesh &, const Input::Record>(*mesh, *it);
+    	secondary_eq = (*it).factory< AdvectionProcessBase, Mesh &, const Input::Record >(*mesh, *it);
```

Comment:
In fact we should rather rename the class to CouplingTHC, and rename the keys for coupled equations, e.g to:
flow,  solute, heat. We have to postpone general coupling schemes till we solve passing the velocity through fields only.